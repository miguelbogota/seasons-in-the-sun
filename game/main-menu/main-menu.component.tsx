'use client';

import { emitCameraControlEvent } from '@app/game/player/camera-control';
import { useGameState } from '@app/game/state';

/**
 * Component with the main menu of the game.
 */
export function MainMenu() {
  const state = useGameState();

  return (
    !state.isPlaying && (
      <div>
        <h1>Seasons in the Sun</h1>

        <button
          onClick={() =>
            state.startGame(() => {
              setTimeout(() => emitCameraControlEvent('lock-controls', null), 200);
            })
          }
        >
          {state.isLoading ? 'Loading...' : 'Begin'}
        </button>
        <button>Options</button>
      </div>
    )
  );
}
