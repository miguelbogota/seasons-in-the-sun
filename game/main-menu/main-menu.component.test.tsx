import { render } from '@testing-library/react';

import { MainMenu } from './main-menu.component';

describe('Game / MainMenu / Component', () => {
  it('should match snapshot', () => {
    const { container } = render(<MainMenu />);
    expect(container).toMatchSnapshot();
  });
});
