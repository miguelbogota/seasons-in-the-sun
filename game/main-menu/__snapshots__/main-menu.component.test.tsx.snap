// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Game / MainMenu / Component > should match snapshot 1`] = `
<div>
  <div>
    <h1>
      Seasons in the Sun
    </h1>
    <button>
      Begin
    </button>
    <button>
      Options
    </button>
  </div>
</div>
`;
